low-level/files/opt/seagate/sspl/setup/sspl_post_install.py

```diff
@@ -0,0 +1,159 @@
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+# GNU Affero General Public License for more details.
+# You should have received a copy of the GNU Affero General Public License
+# along with this program. If not, see <https://www.gnu.org/licenses/>.
+# For any questions about this software or licensing,
+# please email opensource@seagate.com or cortx-questions@seagate.com.
+
+import os
+import errno
+import shutil
+import distutils.dir_util
+from cortx.sspl.bin.sspl_constants import (REPLACEMENT_NODE_ENV_VAR_FILE, PRODUCT_NAME, SSPL_BASE_DIR,
+    file_store_config_path, PRODUCT_BASE_DIR)
+from cortx.utils.process import SimpleProcess
+from cortx.utils.service import Service
+
+
+class PostInstallError(Exception):
+    """ Generic Exception with error code and output """
+
+    def __init__(self, rc, message, *args):
+        """Postinstallerror init."""
+        self._rc = rc
+        self._desc = message % (args)
+
+    def __str__(self):
+        """Postinstallerror str."""
+        if self._rc == 0: return self._desc
+        return "error(%d): %s" %(self._rc, self._desc)
+
+
+class SSPLPostInstall:
+    def __init__(self, args: list):
+        """Ssplpostinstall init."""
+        self.args = args
+        self.name = "sspl_post_install"
+        self._script_dir = os.path.dirname(os.path.abspath(__file__))
+        self.RSYSLOG_CONF="/etc/rsyslog.d/0-iemfwd.conf"
+        self.RSYSLOG_SSPL_CONF="/etc/rsyslog.d/1-ssplfwd.conf"
+        self.PACEMAKER_INSTALLATION_PATH="/lib/ocf/resource.d/seagate/"
+
+    def process(self):
+        if not self.args or len({'-p', '-e', '-c'} & set(self.args)) == 0:
+            raise PostInstallError(errno.EINVAL, f"Invalid Argument for {self.name}")
+
+        PRODUCT = ""
+        ENVIRONMENT = ""
+        RMQ_CLUSTER = "true"
+        i = 0
+        while i < len(self.args):
+            try:
+                if self.args[i] == '-p':
+                    PRODUCT = self.args[i+1]
+                elif self.args[i] == '-e':
+                    ENVIRONMENT = self.args[i+1]
+                elif self.args[i] == '-c':
+                    RMQ_CLUSTER = self.args[i+1]
+            except (IndexError, ValueError):
+                raise PostInstallError(errno.EINVAL, f"Invalid Argument for {self.name}")
+            i+=1
+
+        # sspl_setup_consul script install consul in dev env and checks if consul process is running
+        # on prod. For node replacement scenario consul will not be running on the new node. But,
+        # there will be two instance of consul running on healthy node. When new node is configured
+        # consul will be brought back on it. We are using VIP to connect to consul. So, if consul
+        # is not running on new node, we dont need to error out. So, need to skip this step for
+        # node replacement case
+        # Onward LDR_R2, consul will be abstracted out and it won't exit as hard dependeny of SSPL
+        if PRODUCT_NAME == "LDR_R1":
```

Comment:
Need to avoid LDR in CORTX. Checking if we can use "CORTXr1"

Reply:
@mariyappanp add a TODO until its confirmed then

Reply:
Done.